### PR TITLE
Add application overview end devices panel

### PR DESCRIPTION
**Pull request by ryaplots**
#### Summary

This PR adds the end devices panel to the application overview

#### Changes

- Add end devices table
- Refactor fetch table to be able to use it here
- Use `<ScrollFader />`
- Add periodical device fetch

##### Steps

1. Log into the console running the staging env and see the list of recent devices in the "Recently active" tab

##### Results

Screenshot 2024-06-07 at 10 07 34

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Testing: The steps/process to test this feature are clearly explained including testing for regressions.
- [ ] Infrastructure: If infrastructural changes (e.g., new RPC, configuration) are needed, a separate issue is created in the infrastructural repositories.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [ ] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by ryaplots** [comment]
If `All` is now a link, should it be in the toggle?

**Comment by kschiffer** [comment]
> If `All` is now a link, should it be in the toggle?

No it's fine as it is now.

**Comment by kschiffer** [comment]
Rebased this on top of the latest redesign branch.

**Review by kschiffer — CHANGES_REQUESTED** [review]
I still see some issues with the ellipsis coloring:
image

- The columns can use a better width distribution of maybe 40/40/20.
- Also the scroll fader does not work. Please take a look at the top entities panel in the general overview to see how this should be applied.
- Instead of using refetching the device list, this should be driven by the event stream. But let's exclude this whole auto-updating for the time being.

File: pkg/webui/console/containers/devices-panel/devices-panel.styl [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      top: 0
+      position: sticky
+      background: var(--c-bg-neutral-min)
+      z-index: 1
+
+.scroll-gradient
+  max-height: 23rem
+
+  & > :last-child
+    height: 2rem
```
Comment (kschiffer) [inline pkg/webui/console/containers/devices-panel/devices-panel.styl]: Can be removed if done via ScrollFader.

File: pkg/webui/console/containers/devices-panel/index.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+const DevicesPanel = () => {
+  const navigate = useNavigate()
+  const [active, setActive] = useState('top')
+  const appId = useSelector(selectSelectedApplicationId)
+
+  const handleChange = useCallback(
+    (_, value) => {
+      if (value === 'all') {
+        navigate(`/applications/${appId}/devices`)
```
Comment (kschiffer) [inline pkg/webui/console/containers/devices-panel/index.js]: Please do not navigate programmatically, but add functionality to make a tab work as a plain link as well.

File: pkg/webui/console/store/middleware/logics/devices.js [hunk 3]
```diff
@@ -137,6 +138,38 @@ const getDevicesListLogic = createRequestLogic({
   },
 })
 
+const getDevicesPeriodicallyLogic = createLogic({
```
Comment (kschiffer) [inline pkg/webui/console/store/middleware/logics/devices.js]: Please see my other comment on removing this periodic fetch for the time being (and just using the normal get logic instead).

**Review by kschiffer — CHANGES_REQUESTED** [review]

File: pkg/webui/components/scroll-fader/index.js [hunk 4]
```diff
@@ -61,13 +61,13 @@ const ScrollFader = React.forwardRef(
       mutationObserver.observe(container, { attributes: false, childList: true, subtree: false })
 
       handleScroll() // Call once on mount if needed
-      container.addEventListener('scroll', handleScroll)
+      container.addEventListener('wheel', handleScroll)
```
Comment (kschiffer) [inline pkg/webui/components/scroll-fader/index.js]: Let's discuss this before merging.

File: pkg/webui/containers/fetch-table/index.js [hunk 5]
```diff
@@ -84,6 +84,8 @@ const FetchTable = props => {
     getItemsAction,
     baseDataSelector,
     className,
+    headerClassName,
+    periodical,
```
Comment (kschiffer) [inline pkg/webui/containers/fetch-table/index.js]: I guess this can be removed then for now?

**Review by kschiffer — APPROVED** [review]

**Pull request merged**